### PR TITLE
Update slashing conditions according to specs

### LINKED ISSUE #1190: Update Slashing Conditions per Spec
Update [IsDoubleVote](https://github.com/prysmaticlabs/prysm/blob/master/beacon-chain/core/attestations/attestation.go#L81) and [IsSurroundVote](https://github.com/prysmaticlabs/prysm/blob/master/beacon-chain/core/attestations/attestation.go#L102) to the latest spec as referenced below. Instead of checking for slot number, we should be checking for epoch number. To complete this issue, we should fix the code and the exisiting unit tests.

Reference: https://github.com/ethereum/eth2.0-specs/pull/373

### PR DESCRIPTION
**Pull request by arku**
Resolves #1190

---

# Description

Updates `IsDoubleVote` and `IsSurroundVote` functions according to the latest specs.

Ref: https://github.com/ethereum/eth2.0-specs/pull/373

### CODE REVIEW

File: beacon-chain/core/attestations/attestation_test.go [hunk 1]
```diff
@@ -22,17 +22,17 @@ func TestIsDoubleVote(t *testing.T) {
 	}
 
 	att2 := &pb.AttestationData{
-		Slot: 2,
+		Slot: 64,
 	}
 
 	if IsDoubleVote(att1, att2) {
-		t.Error("It is a double vote despite the attestations being on different slots")
+		t.Error("It is a double vote despite the attestations being on different slot epochs")
```
Comment (nisdas) [inline beacon-chain/core/attestations/attestation_test.go]: should be epochs not slot epochs

File: beacon-chain/core/attestations/attestation_test.go [hunk 2]
```diff
@@ -22,17 +22,17 @@ func TestIsDoubleVote(t *testing.T) {
...
 	}
 
 	att2.Slot = 1
 
 	if !IsDoubleVote(att1, att2) {
-		t.Error("It is not a double vote despite the attestations being on the same slot")
+		t.Error("It is not a double vote despite the attestations being on the same slot epoch")
```
Comment (nisdas) [inline beacon-chain/core/attestations/attestation_test.go]: same here

File: beacon-chain/core/attestations/attestation_test.go [hunk 3]
```diff
@@ -48,18 +48,17 @@ func TestIsSurroundVote(t *testing.T) {
...
 	if IsSurroundVote(att1, att2) {
-		t.Error("It is a surround vote despite both attestations having the same justified slot")
+		t.Error("It is a surround vote despite both attestations having the same justified slot epoch")
 	}
 
-	att2.Slot++
 
 	if IsSurroundVote(att1, att2) {
-		t.Error("It is a surround vote despite both attestations having the same slot.")
+		t.Error("It is a surround vote despite both attestations being on the same slot epoch")
```
Comment (nisdas) [inline beacon-chain/core/attestations/attestation_test.go]: same here

File: beacon-chain/core/attestations/attestation_test.go [hunk 4]
```diff
@@ -48,18 +48,17 @@ func TestIsSurroundVote(t *testing.T) {
 	}
 
 	if IsSurroundVote(att1, att2) {
-		t.Error("It is a surround vote despite both attestations having the same justified slot")
+		t.Error("It is a surround vote despite both attestations having the same justified slot epoch")
```
Comment (nisdas) [inline beacon-chain/core/attestations/attestation_test.go]: same here

**Comment by arku** [comment]
@nisdas Thanks for the review!

I have addressed your comments. Let me know if I missed something.

/cc @prestonvanloon

**Pull request merged**